app/models/user.rb

```diff
@@ -9,4 +9,8 @@ class User < ActiveRecord::Base
   has_many :requests, foreign_key: 'requester_id', class_name: 'Request'
   has_many :offers, foreign_key: 'helper_id', class_name: 'Offer'
   has_many :quests, foreign_key: 'helper_id', class_name: 'Request'
+
+  def reward_karma_points(points)
```

Comment:
i think i would prefer to keep the karma calculations in the karma controller...

Reply:
It would make more sense to actually move the karma subtraction method from the controller to the user as well. That method really doesn't belong in a controller and as the user is the model with the karma attribute, it should be responsible for editing its own attributes.

Reply:
So I think we need to agree on keeping the karma controller at all?

Reply:
I think keeping it gives an easy way to get a user's karma points. It is less complicated than hijacking the /user at least, which likely is the alternative.

Reply:
Okay 👍